bucket_map/src/bucket_storage.rs

```diff
@@ -164,16 +160,16 @@ impl BucketStorage {
         let ix = (ix * self.cell_size) as usize;
         //debug!("FREE {} {}", ix, uid);
         let hdr_slice: &[u8] = &self.mmap[ix..ix + std::mem::size_of::<Header>()];
-        let mut e = Err(BucketStorageError::InvalidFree);
         unsafe {
             let hdr = hdr_slice.as_ptr() as *const Header;
             //debug!("FREE uid: {}", hdr.as_ref().unwrap().uid());
-            if hdr.as_ref().unwrap().unlock(uid) {
-                self.used.fetch_sub(1, Ordering::Relaxed);
-                e = Ok(());
-            }
-        };
-        e
+            let previous_owner = hdr.as_ref().unwrap().unlock();
+            assert_ne!(
+                previous_owner, 0,
+                "free: unlocked a header that was already unlocked"
+            );
```

Comment:
Since errors from `free()` were always unwrapped, change that to an assert here. There's no recovery built in anyway if `free()` were to fail.

Reply:
As per our call, I've fixed this assert.